### PR TITLE
Add friend group

### LINKED ISSUE #13: Add Friend Group
As an orderly user, I can create friend groups and add people into the groups so that I can customize my own categories of friends for better management.

### PR DESCRIPTION
**Pull request by Assyarul**
There is no clear way of organizing contacts into separate groups.

Let's add the feature in.

Fixes #13

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddGroupCommand.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import seedu.address.model.group.Group;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+
+public class AddGroupCommand extends Command {
+    public static final String COMMAND_WORD = "add-group";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds the group of the specified name with the"
+            + "specified persons indicated by the index numbers used in the last person listing. If group exists,"
+            + "add people into group if they were not in it.  \n"
```
Comment (ivantjh) [inline src/main/java/seedu/address/logic/commands/AddGroupCommand.java]: I think this should be in the present tense. `if they are not in it`

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -130,4 +132,21 @@ public static Tag parseTag(String tag) throws ParseException {
...
+    /**
+     * Parses a {@code String indexes} into a {@code List<Index>}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code address} is invalid.
+     */
+    public static List<Index> parseIndexes(String indexes) throws ParseException {
+        requireNonNull(indexes);
+        String[] indexStringArray = indexes.trim().split(" ");
+        List<Index> indexArray = new ArrayList<>();
```
Comment (ivantjh) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:161]: Can be shortened to 
```java
return Arrays.stream(arr)
  .map(ParserUtil::parseIndex)
  .collect(Collectors.toList());
```

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -108,10 +113,32 @@ public void addPerson(Person person) {
...
     public void setPerson(Person target, Person editedPerson) {
         requireAllNonNull(target, editedPerson);
-
         addressBook.setPerson(target, editedPerson);
     }
 
+    @Override
+    public void addGroup(Group group) {
+        addressBook.addGroup(group);
+        updateFilteredPersonList(x->group.getPersons().contains(x));
```
Comment (ivantjh) [inline src/main/java/seedu/address/model/ModelManager.java]: I think there should be a space between the arrow heads. `updateFilteredPersonList(x -> group.getPersons().contains(x))`

File: src/main/java/seedu/address/storage/JsonAdaptedGroup.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    private final String groupName;
+    private final List<String> personNames = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedPerson} with the given meeting details.
+     */
+    @JsonCreator
+    public JsonAdaptedGroup(@JsonProperty("name") String groupName,
+                            @JsonProperty("persons")List<String> personNames) {
```
Comment (ivantjh) [inline src/main/java/seedu/address/storage/JsonAdaptedGroup.java]: Wrong indentation, change intellij settings

File: src/main/java/seedu/address/storage/JsonAdaptedGroup.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                .map(Person::getName).map(Name::toString)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted group object into the model's {@code Group} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted group.
+     */
+    public Group toModelType(List<Person> personList) throws IllegalValueException {
```
Comment (ivantjh) [inline src/main/java/seedu/address/storage/JsonAdaptedGroup.java:55]: Having a parameter here is out of the norm. Can add a comment here saying that it requires the actual personList serialised from the `persons` json file?

File: src/main/java/seedu/address/storage/JsonAdaptedGroup.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        final Name modelName = new Name(groupName);
+
+        final Set<Person> modelPersonSet = new HashSet<>();
+        for (String personName : personNames) {
+            if (!Name.isValidName(personName)) {
+                throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+            }
+            Person person;
+            person = personList.stream().filter(x->x.getName().toString().equals(personName)).findFirst().get();
```
Comment (ivantjh) [inline src/main/java/seedu/address/storage/JsonAdaptedGroup.java]: Multi-line for readability. I think the lambda has a spaces enclosing the arrowhead.

**Review by ivantjh — APPROVED** [review]
Minor styling errors. Looks good to merge! 🎈

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 2 above)
Comment (Assyarul) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:161]: Tried using stream but parseIndex throws an exception. Apparently unhandled exception does poorly with streams unless I explicitly throw it inside the lambda.

**Pull request merged**